### DISCUSSION TITLE
Calling other docker containers with Airflow running in its own docker container

### DISCUSSION THREAD

**Discussion by YerePhy** [apache/airflow#25020] [Q&A] [answered]
Hi, 

I'm new to Airflow and Docker and I would like to know if it is a good practice to call a docker container from the container where Airflow is runing and what it the recommended way of doing this. My (long term) goal is to learn how create a data processing/prediction pipeline and maybe enhance it in the future for automatic model training. Since I am new to these technologies, a high level comprehensive explanation with maybe some guidelines would be really really really helpful. 

I'm asking this from ignorace, sorry if I am not being specific enough.

Thanks in advance.

**Comment by potiuk** [comment, marked as answer]
You are lookint at something called docker-in-docker : https://devopscube.com/run-docker-in-docker/ is an example of it but you can search in general for dind or docker-in-docker solution. This is nothing Airlfow-Specific, so there are no airlfow-specific quidelines here. Thera are plenty of resources available and a number of tutorials onlne that you can explore.

**Reply by YerePhy** [reply]
Hi @potiuk,

Thanks for your answer, I'm a little bit lost... would you say that it is a common/good practice to use Airflow in a docker in docker set up? Or the common practice is to use Airflow installed directly in the host. I'm trying to find what is the role of `DockerOperator` if you are running Airflow inside a Docker container... just to have a mental picture and see if Airflow can benefit my development somehow.

**Reply by potiuk** [reply]
It's up to you how you install it. You can see the various ways you can install and use airlfow here https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html and you can assess yourself which way is best.

**Reply by potiuk** [reply]
There is no "beter" or "worse" or "common" and "less common" - you have to decide what's best for you yourself. You are the only person in the world who know your circumstances and what will work for you (and the docs has all the informtion that allows you to asses what is best in your case).

**Reply by YerePhy** [reply]
Okay, thanks

**Reply by YerePhy** [reply]
Yes, you're right. But I just looking for help because most tutorials I've seen just give you an overview and basic usage of Airflow. I'm learning Docker as well and I tought that maybe the community could help in understanding which are the recommended set ups when you need to call external containers from a the container where Airflow is running. Thanks for your references, wish me luck ;)
